pkgs/by-name/ta/taler-wallet-core/package.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitHub,
+  fetchgit,
+  git,
+  jq,
+  nodejs,
+  python3,
+  zip,
+  pnpm,
+  removeReferencesTo,
+  srcOnly,
+}:
+let
+  nodePackages = nodejs.pkgs;
+  nodeSources = srcOnly nodejs;
+  esbuild_0_19_9 = buildGoModule rec {
+    pname = "esbuild";
+    version = "0.19.9";
+
+    src = fetchFromGitHub {
+      owner = "evanw";
+      repo = "esbuild";
+      rev = "v${version}";
+      hash = "sha256-GiQTB/P+7uVGZfUaeM7S/5lGvfHlTl/cFt7XbNfE0qw=";
+    };
+    vendorHash = "sha256-+BfxCyg0KkDQpHt/wycy/8CTG6YBA/VJvJFhhzUnSiQ=";
+
+    subPackages = [ "cmd/esbuild" ];
+
+    ldflags = [
+      "-s"
+      "-w"
+    ];
+
+    meta = {
+      description = "An extremely fast JavaScript bundler";
+      homepage = "https://esbuild.github.io";
+      license = lib.licenses.mit;
+      mainProgram = "esbuild";
+    };
+  };
+  customPython = python3.withPackages (p: [ p.setuptools ]);
+in
+stdenv.mkDerivation rec {
+  pname = "taler-wallet-core";
+  version = "0.12.6";
+
+  src = fetchgit {
+    url = "https://git.taler.net/wallet-core.git";
+    rev = "v${version}";
+    hash = "sha256-gZfXzw3B1Ei2F5o07bpPm6o2alcZ41R6OyhSpew0eTA=";
+  };
+
+  nativeBuildInputs = [
+    git
+    jq
+    nodePackages.nodejs
+    pnpm.configHook
+    customPython
+    zip
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-OwZLRv2ZWJ01pHWS+CGxvM4J19tGnEsRJ9S3bcsaWSE=";
+  };
+
+  buildInputs = [ nodePackages.nodejs ];
+
+  # Use a fake git?
+  postUnpack = ''
+    git init -b master
+    git config user.email "root@localhost"
+    git config user.name "root"
+    git commit --allow-empty -m "Initial commit"
+  '';
```

Comment:
Why is this needed?

Reply:
I suppose because the package does not build if it's not in a git dir. Here is what I get when I remove this:

```
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set)
```

Reply:
I suggest you try `touch .git`.

Reply:
```
fatal: invalid gitfile format: /build/.git
```

Reply:
Not only does the program want a git repo, but it also wants a commit to be present (perhaps it uses it for something?). So I don't think there is any way around doing this.